modules/context/repo.go

```diff
@@ -188,7 +188,11 @@ func RetrieveBaseRepo(ctx *Context, repo *models.Repository) {
 
 // ComposeGoGetImport returns go-get-import meta content.
 func ComposeGoGetImport(owner, repo string) string {
-	return path.Join(setting.Domain, setting.AppSubURL, url.PathEscape(owner), url.PathEscape(repo))
+	host := setting.Domain
+	if (setting.Protocol == setting.HTTP && setting.HTTPPort != "80") || (setting.Protocol == setting.HTTPS && setting.HTTPPort != "443") {
```

Comment:
If Gitea is running behind reverse proxy this will be incorrect imho

Reply:
So you think we should just use url.Parse(setting.AppURL) and grab it from there?

Reply:
No I knew I'd checked this:

https://github.com/go-gitea/gitea/blob/355ab0c62c1d58200d68e30d112db398ea00297a/modules/setting/setting.go#L583-L592

This will be updated with the correct suburl

Reply:
OK, I think setting.Protocol can't necessary be trusted to be updated - therefore I've switched to parsing the AppURL.